sdk/ml/azure-ai-ml/azure/ai/ml/entities/_inputs_outputs/output.py

```diff
@@ -73,9 +73,10 @@ def __init__(self, *, type=AssetTypes.URI_FOLDER, path=None, mode=None, descript
         # As an annotation, it is not allowed to initialize the name.
         # The name will be updated by the annotated variable name.
         self.name = None
+        self.asset_name = kwargs.pop('asset_name', None)
```

Comment:
IIRC, as per Qianqian Nie, we should not accept version if name is not available. Please check with Qianqian .

Reply:
correct, we should validate version is only acceptable when name is provided

Reply:
I add test_register_output_without_name_cli and test_register_output_without_name_sdk in pipeline_job/e2etests/test_piipeline_job.py